pkg/kubelet/pluginmanager/cache/desired_state_of_world_test.go

```diff
@@ -70,6 +71,10 @@ func Test_DSW_AddOrUpdatePlugin_Positive_ExistingPlugin(t *testing.T) {
 		t.Fatalf("Expected\n%s\nin desired state of world, but got\n%v\n", socketPath, dswPlugins[0])
 	}
 	oldTimestamp := dswPlugins[0].Timestamp
+	// On Windows, time.Now() is not as precise, which means that 2 consecutive calls may
+	// return the same timestamp. This would cause the test to fail, since they would have
+	// the same timestamp. Sleeping will cause in different times.
+	time.Sleep(time.Millisecond)
```

Comment:
is this a real runtime bug in Windows environments we should fix, rather than working around it in the test? e.g. if we observe two changes and call AddOrUpdatePlugin within the same millisecond, will that result in identical timestamps that make subsequent uses of the timestamp fail? do we need to inject a millisecond wait for windows in the update path in `AddOrUpdatePlugin` instead of working around the issue here?

Reply:
``AddOrUpdatePlugin`` doesn't do much in the first place. It simply adds a new ``PluginInfo`` into its ``dsw.socketFileToInfo`` map based on the given socket path. If an entry in the map with that socket path already exists, it will just log a message, and then proceed to create a new ``PluginInfo`` with a new timestamp, replacing the previous one. Because of Windows, these 2 timestamps happen to be the same, being taken into quick succession, which results in the ``if !newDswPlugins[0].Timestamp.After(oldTimestamp)`` assertion to fail.

Basically, we're sleeping 1 millisecond here in order to give the clock time to update, to have a different value. The workaround here is different from the ``actual_state_of_world_test.go`` change because the ``AddOrUpdatePlugin`` method itself creates the ``PluginInfo`` object with ``time.Now()`` timestamp, and it's not something set by the test itself.

Alternatively, we could update the ``dswPlugins[0].Timestamp`` ourselves to one 1 millisecond before, which means that we don't have to sleep in order to have a newer timestamp for the updated plugin to pass the test.